FoundationsMastery/CustomString.cs

```diff
@@ -0,0 +1,75 @@
+﻿using FoundationsMastery.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FoundationsMastery
+{
+    public class CustomString : ICustomString
+    {
+        public CustomString()
+        {
+        }
+
+        public CustomString(IEnumerable<char> contents)
+        {
+            this.Contents = contents;
+        }
+
+        public IEnumerable<char> Contents { get; set; } 
+       
+        public int Length {
+            get
+            {
+                return this.Contents.Count(); 
+            }
+        } 
+
+        public void Clear() 
+        {
+            Contents = new List<char>(); 
+            Contents.ToList().Clear(); 
+        }   
+
+        public string Concat(IEnumerable<char> rhs) 
+        {
+            string concat = "";
+            foreach (char a in this.Contents) 
+            {
+                concat += a;
+            }
+            foreach (char a in rhs) 
+            {
+                concat += a;
+            }
+            return concat;
```

Comment:
Looping over each one does work - but is there a way we can treat both of them as strings (`Contents` and `rhs`)? It's less work than building them one character at a time